br/pkg/lightning/backend/local/region_job.go

```diff
@@ -202,6 +202,33 @@ func (local *Backend) writeToTiKV(ctx context.Context, j *regionJob) error {
 	return err
 }
 
+func getWriteMaxSize(j *regionJob) int64 {
+	// if region-split-size <= 96MiB, we bump the threshold a bit to avoid too many retry split
+	// because the range-properties is not 100% accurate
+	maxSize := j.regionSplitSize
+	if j.regionSplitSize <= int64(config.SplitRegionSize) {
+		maxSize = j.regionSplitSize * 4 / 3
+	}
+	return maxSize
+}
+
+const defaultWriteTimeout = 15 * time.Minute
+
+func getWriteMaxTimeout(j *regionJob, limiter StoreWriteLimiter) time.Duration {
+	switch v := limiter.(type) {
+	case noopStoreWriteLimiter:
+		return defaultWriteTimeout
+	case *storeWriteLimiter:
+		maxSize := getWriteMaxSize(j)
+		// limit is not 0, otherwise local backend will use noopStoreWriteLimiter
+		durationSec := maxSize / int64(v.limit)
```

Comment:
this value is a few seconds in most cases, seems not that useful to adjust `timeout`